microraiden/microraiden/client/http_client.py

```diff
@@ -106,7 +109,7 @@ def _request_resource(self, requested_resource: str) -> (Any, bool):
                 )
             return None, retry
         else:
-            return None, False
+            return None, True
```

Comment:
Under the new default behavior this code isn't reached, right?

If someone decides to override the check in `on_http_response()`, then this code would be reachable though. Should we really continue on errors like 500?

Maybe we should add another `on_server_error()` callback here which defaults to `True`, just to give users a maximum amount of configurability with sensible defaults.

Reply:
There'll always be weird use cases, for example a scenario where user will want to have a simple HTTP auth to the client (for one reason or another), then the sequence will be 401 -> 402 -> 200.
It's anyways user's responsibility to recover if he is redefining this method..